care/emr/resources/scheduling/schedule/spec.py

```diff
@@ -77,8 +78,15 @@ def validate_for_slot_type(self):
                 end_time = datetime.datetime.combine(
                     datetime.datetime.now(tz=UTC).date(), availability.end_time
                 )
+                availability_duration_in_seconds = (
+                    end_time - start_time
+                ).total_seconds()
                 slot_size_in_seconds = self.slot_size_in_minutes * 60
-                if (end_time - start_time).total_seconds() % slot_size_in_seconds != 0:
+                total_slots = availability_duration_in_seconds / slot_size_in_seconds
+                if total_slots > settings.MAX_SLOTS_PER_AVAILABILITY:
+                    error_message = f"Daily slot allocation limit exceeded; maximum allowed is {settings.MAX_SLOTS_PER_AVAILABILITY} slots per day."
```

Comment:
Error message needs to be refined. It's per availability, not per day.

I could create multiple availabilities that'd exceed this.

```suggestion
                    error_message = f"Too many slots per availability. Maximum allowed is {settings.MAX_SLOTS_PER_AVAILABILITY} slots per availability session."
```